test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/Utilities/TestExtensionsTests.cs

```diff
@@ -239,4 +241,110 @@ public void GetExtensionsDiscoveredFromAssemblyShouldReturnTestDiscoveresAndLogg
         };
         CollectionAssert.AreEqual(expectedLoggers.Keys, extensions.TestLoggers.Keys);
     }
+
+    [TestMethod]
+    public void MergedDictionaryOfEmptyDictionariesShouldBeAnEmptyDictionary()
+    {
+        var first = new Dictionary<string, HashSet<string>>();
+        var second = new Dictionary<string, HashSet<string>>();
+        var merged = TestExtensions.CreateMergedDictionary(first, second);
+
+        // Merging two empty dictionaries should result in an empty dictionary.
+        Assert.IsTrue(merged.Count == 0);
+
+        // Make sure the method is "pure" and returns a new reference.
+        Assert.IsFalse(ReferenceEquals(merged, first));
+        Assert.IsFalse(ReferenceEquals(merged, second));
+    }
+
+    [TestMethod]
+    public void MergedDictionaryOfOneEmptyAndOneNotEmptyDictionaryShouldBeNotEmpty()
+    {
+        var first = new Dictionary<string, HashSet<string>>();
+        var second = new Dictionary<string, HashSet<string>>
+        {
+            { "aaa", new HashSet<string>() }
+        };
+
+        var merged1 = TestExtensions.CreateMergedDictionary(first, second);
+        var merged2 = TestExtensions.CreateMergedDictionary(second, first);
+
+        // Merging one empty dictionary with a not empty one should result in a not empty
+        // dictionary.
+        Assert.IsTrue(merged1.Count == 1);
+        Assert.IsTrue(merged2.Count == 1);
+        Assert.IsTrue(merged1.ContainsKey("aaa"));
+        Assert.IsTrue(merged2.ContainsKey("aaa"));
+
+        // Make sure the method stays "pure" and returns a new reference regardless of the input.
+        Assert.IsFalse(ReferenceEquals(merged1, first));
+        Assert.IsFalse(ReferenceEquals(merged1, second));
+        Assert.IsFalse(ReferenceEquals(merged2, first));
+        Assert.IsFalse(ReferenceEquals(merged2, second));
+        Assert.IsFalse(ReferenceEquals(merged1, merged2));
+    }
+
+    [TestMethod]
+    public void MergedDictionaryShouldBeSuccessful()
```

Comment:
BeSuccessful says nothing about what the desired result is. 

Also use fluent assertions to do the equality / equivalency check.

Reply:
Done.